src/DynamoRevit/ViewModel/RevitWatch3DViewModel.cs

```diff
@@ -221,7 +223,10 @@ private IEnumerable<GeometryObject> Tessellate(Curve curve)
             {
                 var xyz0 = new XYZ(verts[i] * conv, verts[i + 1] * conv, verts[i + 2] * conv);
                 var xyz1 = new XYZ(verts[i + 3] * conv, verts[i + 4] * conv, verts[i + 5] * conv);
-
+                if (xyz0.DistanceTo(xyz1) < DocumentManager.Instance.CurrentUIApplication.Application.ShortCurveTolerance)
+                {
+                    continue;
+                }
```

Comment:
Yes, @mjkkirschner has got what I have suggested.

Just to make it more clear:
1). Check the curve's length at the very beginning of the function, if it is less than the tolerance, return from there;
2). Facet the curve;
3). Create one polyline from the faceted points;